### ISSUE TITLE
C# Extension Plugin

### ISSUE THREAD

**Issue by sraedler** [Kooboo/Kooboo#22]
Dear KobooTeam,

is it possible, that you provide an example of an C# extension or a template?
Is it possible to use the old one (Old Kooboo)?

Thanks

**Comment by koobooteam** [comment]
There are many things you can do with extension.

You can find some examples here:
https://www.kooboo.cn/en/documentation/csharp-extensions

If you know what you want to do with your extension, I might have a better idea of how to provide an example for you.

**Comment by sraedler** [comment]
Could you provide a template csproj ?
It could be something like a random number generation, which is displayed on a View

**Comment by koobooteam** [comment]
To make a template, you do not a csproj any more. 

You can make any website in Kooboo, one click to export them as templates, that can be imported later or share.

There are a few coding template websites at:
https://www.kooboo.cn/en/documentation/kscript_example

**Comment by koobooteam** [comment]
www.kooboo.com & www.kooboo.cn are two identical websites. , but I am moving kooboo.com to new server tonight, so I give you links to www.kooboo.cn

**Comment by sraedler** [comment]
I saw that kscript. But I would prefer to write some C# extensions.

Is there any example?
Additionally how to debug it?

**Comment by sraedler** [comment]
Maybe I my request was a little bit unclear.

I don't want to make a template. I want to write a C# Plugin.
This should be done in MVC5 like it was possible in the old Kooboo.

Is it possible here again?

**Comment by koobooteam** [comment]
Yes, we do have a module development Visual Studio Template. But I haven't reviewed it recently since we change a lot of our API now. 

In this Vs project, you hook menu, pages and api into the CMS both backend & frond end. Our future development will be done in this way as well. 

I kindly of lack time to clean the code base and make it public.

**Comment by sraedler** [comment]
Good to hear!

thanks in advance.
